tests/framework/microvm.py

```diff
@@ -333,10 +335,22 @@ def spawn(self):
 
             run(start_cmd, shell=True, check=True)
 
-            out = run('screen -ls', shell=True, stdout=PIPE)\
-                .stdout.decode('utf-8')
-            screen_pid = re.search(r'([0-9]+)\.{}'.format(self._session_name),
-                                   out).group(1)
+            # Build a regex object to match (number).session_name
+            regex_object = re.compile(
+                r'([0-9]+)\.{}'.format(self._session_name))
+
+            # Check the output of 'screen -ls' until we get the PID or
+            # a timeout
```

Comment:
What is this timeout referring to? It is the timeout specified in `pytest.ini`?

Reply:
It refers to an exception thrown by the retry() call. I'll reword the comment.